src/java/ognl/security/MethodBodyExecutionSandbox.java

```diff
@@ -0,0 +1,158 @@
+package ognl.security;
+
+import java.lang.reflect.InvocationTargetException;
+import java.lang.reflect.Method;
+import java.security.Permissions;
+import java.security.Policy;
+import java.security.SecurityPermission;
+
+/**
+ * Guarantee a singleton shared thread-safe security manager and sandbox for user's methods body execution
+ * @author Yasser Zamani
+ * @since 3.1.23
+ */
+public class MethodBodyExecutionSandbox {
```

Comment:
I think that these flag and permission variables either need to be made volatile, or all the access points to them may need to by synchronized.

Reply:
deleted due to not being meaningful enough. Will be added in future if user demand.